source/bao_hyp/config.rst

```diff
@@ -65,41 +65,512 @@ describe the image running on the VM (see `Guest Image`_);
 2. A C string with the guest image's binary file path. It can be either an absolute path or a path
 relative to the config source file.
 
-Guests Configuration
+VM Configuration
 --------------------
 
+Bao's configuration file allows you to partition the platforms' hardware resources, such as CPU
+cores, memory, or devices, by assigning them to one or more VMs. It also allows you to configure
+the guest image to run on that VM. In Bao, resources are exclusively assigned to each VM, including
+communication channels between two or more VMs, which may utilize shared memory or dedicated links.
+
+.. figure:: img/guest-config.svg
+    :align: center
+    :name: guest-config-fig
+
+|
+
+Each entry in the ``vmlist`` mentioned earlier is a ``vm_config`` struct, which defines the
+configuration of each individual guest:
+
+.. code-block:: c
+
+    struct vm_config {
+        struct {
+          vaddr_t base_addr;
+          paddr_t load_addr;
+          size_t size;
+          bool separately_loaded;
+          bool inplace;
+        } image;
+        vaddr_t entry;
+        cpumap_t cpu_affinity;
+        colormap_t colors;
+        struct vm_platform platform;
+    };
+
+Each entry in this list represents a unique VM configuration, defining its image, memory address,
+CPU affinity, color mapping, and platform details. For each VM, the following parameters must be
+specified:
+
+- ``image`` [mandatory] - a structure containing information about guest image loading (see details
+  in `Guest Image`_)
+- ``entry`` [mandatory] - defines the entry point address in guest's address space;
+- ``platform description`` [mandatory] - a description of the VM platform, defining its resource
+  assignments and requirements (see details in `Virtual Machine Configuration`_);
+- ``cpu_affinity`` [optional] - defines the affinity of the VM's vCPUs to the physical CPUs
+  assigned to the virtual platform. The affinity will be followed as best as possible, but it is
+  not guaranteed (see details in `CPU Affinity`_).;
+- ``colors`` [optional] - assignment of shared LLC cache colors (or partitions) to this VM (see
+  details in `Coloring`_).
+
 Guest Image
 ***********
+.. _Guest Image:
+
+The guest ``image`` comprises a structure that describes the image configuration running on the
+guest side. It encompasses the following options:
+
+- **image** [mandatory] - definition of the ``image`` to run on a given VM. The ``image``
```

Comment:
```suggestion
- ``image`` [mandatory] - definition of the ``image`` to run on a given VM. The ``image``
```

Reply:
Thanks! Fixed in https://github.com/bao-project/bao-docs/pull/78/commits/12a303c36d4dc3d6310e65291c125de5729336a5!